File: src/services/apiService.js
const API_BASE_URL = 'http://localhost:8000/api';

const apiService = {
    // Dashboard Metrics
    getDashboardMetrics: async (period = 'Yearly', month, year) => {
        let url = `${API_BASE_URL}/dashboard/metrics?period=${period}`;
        if (month !== undefined) url += `&month=${month}`;
        if (year !== undefined) url += `&year=${year}`;
        const response = await fetch(url);
        if (!response.ok) throw new Error('Failed to fetch dashboard metrics');
        return response.json();
    },

    // Plants
    getPlants: async () => {
        const response = await fetch(`${API_BASE_URL}/plants`);
        if (!response.ok) throw new Error('Failed to fetch plants');
        return response.json();
    },

    getPlantDetails: async (plantId) => {
        const response = await fetch(`${API_BASE_URL}/plants/${plantId}`);
        if (!response.ok) throw new Error('Failed to fetch plant details');
        return response.json();
    },

    // Inverters / Devices
    getDevices: async () => {
        const response = await fetch(`${API_BASE_URL}/devices`);
        if (!response.ok) throw new Error('Failed to fetch devices');
        return response.json();
    },

    getDeviceDetails: async (deviceId) => {
        const response = await fetch(`${API_BASE_URL}/devices/${deviceId}`);
        if (!response.ok) throw new Error('Failed to fetch device details');
        return response.json();
    },

    getDeviceTelemetry: async (deviceId) => {
        const response = await fetch(`${API_BASE_URL}/devices/${deviceId}/telemetry`);
        if (!response.ok) throw new Error('Failed to fetch telemetry');
        return response.json();
    },

    // Alerts
    getAlerts: async () => {
        const response = await fetch(`${API_BASE_URL}/alerts`);
        if (!response.ok) throw new Error('Failed to fetch alerts');
        return response.json();
    },

    // AI Diagnosis
    diagnoseDevice: async (deviceId) => {
        const response = await fetch(`${API_BASE_URL}/ai/diagnose/${deviceId}`, {
            method: 'POST'
        });
        if (!response.ok) throw new Error('Failed to run AI diagnosis');
        return response.json();
    }
};

export default apiService;

File: solaris_flutter/lib/services/api_service.dart
import 'package:dio/dio.dart';
import '../core/api/api_client.dart';
import '../models/dashboard_metrics.dart';
import '../models/plant_model.dart';
import '../models/device_model.dart';
import '../models/alert_model.dart';
import '../models/voice_model.dart';

class ApiService {
  final ApiClient _apiClient = ApiClient();
  Dio get _dio => _apiClient.dio;

  // --- Dashboard ---
  Future<DashboardMetrics> getDashboardMetrics() async {
    final response = await _dio.get('/api/dashboard/metrics');
    return DashboardMetrics.fromJson(response.data);
  }

  // --- Plants ---
  Future<List<PlantSummary>> getPlants() async {
    final response = await _dio.get('/api/plants');
    final List<dynamic> data = response.data;
    return data.map((json) => PlantSummary.fromJson(json)).toList();
  }

  Future<PlantDetail> getPlant(String id) async {
    final response = await _dio.get('/api/plants/$id');
    return PlantDetail.fromJson(response.data);
  }

  // --- Devices ---
  Future<List<DeviceSummary>> getDevices() async {
    final response = await _dio.get('/api/devices');
    final List<dynamic> data = response.data;
    return data.map((json) => DeviceSummary.fromJson(json)).toList();
  }

  Future<DeviceSummary> getDevice(String id) async {
    final response = await _dio.get('/api/devices/$id');
    return DeviceSummary.fromJson(response.data);
  }

  Future<List<TelemetryReading>> getDeviceTelemetry(String id) async {
    final response = await _dio.get('/api/devices/$id/telemetry');
    final List<dynamic> data = response.data;
    return data.map((json) => TelemetryReading.fromJson(json)).toList();
  }

  Future<List<CTAnalysisData>> getCTAnalysis(String id) async {
    final response = await _dio.get('/api/devices/$id/ct-analysis');
    final List<dynamic> data = response.data;
    return data.map((json) => CTAnalysisData.fromJson(json)).toList();
  }

  Future<VoltageGridData> getVoltageGrid(String id) async {
    final response = await _dio.get('/api/devices/$id/voltage-grid');
    return VoltageGridData.fromJson(response.data);
  }

  // --- Alerts ---
  Future<List<AlertOut>> getAlerts() async {
    final response = await _dio.get('/api/alerts');
    final List<dynamic> data = response.data;
    return data.map((json) => AlertOut.fromJson(json)).toList();
  }

  Future<void> acknowledgeAlert(int id) async {
    await _dio.post('/api/alerts/$id/acknowledge');
  }

  // --- AI Diagnosis ---
  Future<DiagnoseResponse> diagnoseDevice(String deviceId, {int? alertId}) async {
    final response = await _dio.post(
      '/api/ai/diagnose',
      data: {
        'device_id': deviceId,
        if (alertId != null) 'alert_id': alertId,
      },
    );
    return DiagnoseResponse.fromJson(response.data);
  }

  // --- Voice Navigation ---
  Future<VoiceCommandResponse> sendVoiceCommand({
    String? audioB64,
    String? textInput,
    String? sessionId,
    required String currentView,
  }) async {
    final response = await _dio.post(
      '/api/voice/command',
      data: {
        if (audioB64 != null) 'audio_b64': audioB64,
        if (textInput != null) 'text_input': textInput,
        if (sessionId != null) 'session_id': sessionId,
        'current_view': currentView,
      },
    );
    return VoiceCommandResponse.fromJson(response.data);
  }
}
